ringpop.go

```diff
@@ -347,6 +356,17 @@ func (rp *Ringpop) LookupN(key string, n int) []string {
 	return rp.ring.LookupN(key, n)
 }
 
+func (rp *Ringpop) LookupReplicationFactor(key string, n ReplicationFactor) []string {
```

Comment:
Instead what if we were to expose `LookupAll()` and later `LookupQuorum()` functions? Having `Lookup()`, `LookupN()`, and now `LookupReplicationFactor()` (that internally uses a set of consts to do the dispatching) is not as straight-forward an API.